internal/gatewayapi/translator.go

```diff
@@ -1083,6 +1083,14 @@ func (t *Translator) ProcessHTTPRoutes(httpRoutes []*v1beta1.HTTPRoute, gateways
 							})
 						}
 					}
+					for _, queryParamMatch := range match.QueryParams {
+						if QueryParamMatchTypeDerefOr(queryParamMatch.Type, v1beta1.QueryParamMatchExact) == v1beta1.QueryParamMatchExact {
+							irRoute.QueryParamMatches = append(irRoute.QueryParamMatches, &ir.StringMatch{
+								Name:  queryParamMatch.Name,
+								Exact: StringPtr(queryParamMatch.Value),
+							})
```

Comment:
I didn't handle the regex match type here because the path and header neither did than. But I don't know why we don't hanle regex.

Reply:
can you raise an issue to track this
cc @skriss

Reply:
https://github.com/envoyproxy/gateway/issues/608